dgraph/graphql/src/resolvers.js

```diff
@@ -1,39 +1,14 @@
-import { UserInputError } from 'apollo-server-fastify';
-import GraphQLJSON from 'graphql-type-json';
+import { queries } from './dgraph/queries';
 
 const resolvers = {
   Query: {
-    searchByName: async (_source, _args, { dataSources }) => {
-      const dgraphQuery = `{
-        query(func: regexp(description, /^${_args.text}/i)) @filter(has(code) AND eq(type, "unit_of_use")) {
-          code
-          description
-        }
-      }`;
+    entities: async (_source, _args, { dataSources }) => {
+      const { filter } = _args;
+      const dgraphQuery = queries.entities(filter);
       const response = await dataSources.dgraph.postQuery(dgraphQuery);
       return response.data.query;
     },
```

Comment:
```suggestion
    entity: async (_source, _args, { dataSources }) => {
      const { code } = _args;
      if (!code) return null;
      const query = queries.entity(code);
      const response = await dataSources.dgraph.postQuery(query);
      return response.data.query;
    },
    entities: async (_source, _args, { dataSources }) => {
      const { filter } = _args;
      const { type } = filter;
      if (!type) return null;
      const query = queries.entities(type);
      const response = await dataSources.dgraph.postQuery(query);
      return response.data.query;
    }
```

Reply:
Thanks for the suggestion! I made code a required parameter in the schema for single entity requests (Apollo will return HTTP 400 / GraphQL validation error) as I think it doesn't make much sense to be searching for a specific entity without one. Aside from that I've pretty much implemented the above as is with https://github.com/openmsupply/unified-codes/commit/b5eebe44775bff017d8b236cbf95e78afbcee3ac